internal/statemachine/image_definition.go

```diff
@@ -0,0 +1,222 @@
+package statemachine
+
+import (
+	"github.com/xeipuuv/gojsonschema"
+)
+
+// ImageDefinition is the parent struct for the data
+// contained within a classic image definition file
+type ImageDefinition struct {
```

Comment:
I like this idea in theory, and yes it is possible to add arbitrary tags to the struct and read them later. I need to think about how it would actually assemble the slice of states, however, as certain options would lead to the states being in a different order. For example, `extra-ppas:` needs to be handled differently if we are building a rootfs vs if we are extracting the rootfs from a tar archive.